app/src/main/java/org/sil/storyproducer/model/StoryIO.kt

```diff
@@ -39,41 +43,54 @@ fun storyFromJson(context: Context, storyTitle: String): Story?{
                 ?: return null
         return adapter.fromJson(fileContents)
     } catch (e: Exception) {
-        return null
+        throw StoryIOException("StoryIO: storyFromJson() parse exception from moshi")
```

Comment:
from what I can tell in https://firebase.google.com/docs/crashlytics/customize-crash-reports?platform=android
about Reporting Non-fatal Exceptions
I think this catch block should do:
FirebaseCrashlytics.getInstance().recordException(e)
    // ...handle the exception. 
   save fileContents in a new file for postmortem analysis
   display alert dialog with message including e.message, button to CONTINUE
return null

Reply:
I like the idea of creating a StoryIOException, but I am not sure what we gain by throwing it to the next level.  I believe we may be actually losing information because "e" has the exact error that occurred during the MOSHI parse. 
(I could be wrong on this one, because of extenuating circumstance I haven't been to check it out).  

> EG: **_com.squareup.moshi.JsonDataException: Expected one of [WORKSPACE, REGISTRATION, STORY_LIST, LEARN, TRANSLATE_REVISE, COMMUNITY_WORK, WHOLE_STORY, BACK_T, REMOTE_CHECK, ACCURACY_CHECK, VOICE_STUDIO, FINALIZE, SHARE] but was DRAFT at path_**

We need to make sure that our exception to FirebaseCrashlytics contains this information because it will eliminate the guess work of what went wrong.  Also, it would be nice to capture the name of the file and method name in the error message back to crashlytics (add to e or create our own string in our own exception) I haven't had enough time to see what is sent to crashlytics (ie, it may contain the name of the routine).  If we return either Story or null, it eliminates several lines of code in the calling program.  See comment below.

I would also propose that we start using the Log Class that is available.  We would be able to filter on log levels in the story producer app when running from Android Studio.  See the Log class for example on how to use.  This may be beyond the scope for the interim, but I think we should start using the Log Class

Reply:
@DaleHensley Yes, we definitely expect that the Crashlytics report will include the detailed message, the kt filename, method, line#. I expect that recordException(e) gives us all that automatically.
Also, FYI issue #544 for SP v3.1 will add logs to the crashlytics reports.